### PR TITLE
Crud functionality branch

### PR DESCRIPTION
**Pull request by baqar-abbas**
- Create Read update and delete functionality added to the To Do List Project.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -36,22 +25,88 @@ btnDeleteAll.setAttribute('class', 'delete-all');
...
 
+const addTask = (text) => {
+  const newTask = {
+    text,
+    completed: false,
+    index: todoItems.length + 1,
+  };
+  todoItems.push(newTask);
+  localStorage.setItem('todoItems', JSON.stringify(todoItems));
+};
```
Comment (anagudelogu) [inline src/index.js]: - [ ] You did a great job working on this milestone! 💯 However, according to the requirements, the new functionality should go inside a separate JavaScript file, please consider doing so! 🤓 🔧 

image

File: src/index.js [hunk 2]
```diff
@@ -36,22 +25,88 @@ btnDeleteAll.setAttribute('class', 'delete-all');
...
+      const itemKey = parseInt(node.getAttribute('data-key'), 10);
+      const itemIndex = todoItems.findIndex((item) => item.index === itemKey);
+      const currentTaskText = todoItems[itemIndex].text; // get the current task text
+      const newTaskText = prompt('Please enter the new task description:', currentTaskText); // pre-fill prompt with current task text
+      if (newTaskText !== null && newTaskText !== '') {
+        todoItems[itemIndex].text = newTaskText;
+        localStorage.setItem('todoItems', JSON.stringify(todoItems));
+        display();
+      }
+    });
```
Comment (anagudelogu) [inline src/index.js]: - [ ] You have very good logic for all the features, kudos for those comments! 🤓 However, please keep in mind that one of the goals of this project is to follow a given project example and try to recreate it as closely as possible!

Please avoid using prompts at all costs, they're impossible to style, stop code execution, and display differently on different browsers. According to the video example, when the user clicks on the three dots, the task description works as an input field and once you press [ENTER] the new description gets saved. Please consider implementing it this way! 🤓 🔧 


Video: https://www.youtube.com/watch?v=AcUd-_Yjjqg

File: src/index.js [hunk 3]
```diff
@@ -36,22 +25,88 @@ btnDeleteAll.setAttribute('class', 'delete-all');
...
+
+// Set up form submit event listener to add a new task
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const inputText = document.querySelector('.inputText');
+  const text = inputText.value.trim();
+  if (text !== '') {
+    addTask(text);
+    inputText.value = '';
+    ul.innerHTML = ''; // clear the list before re-rendering it
```
Comment (anagudelogu) [inline src/index.js]: - [ ] I think that you can safely remove this clear list from this listener, as you already have it at the start of your `display()` function! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @baqar-abbas, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of GitHub flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 3 above)
Comment (baqar-abbas) [reply src/index.js]: In order to implement the edit functionality with out prompt I have used input element which becomes editable on user clicking on edit option screen shot attached for your kind reference 
image

Comment (baqar-abbas) [reply src/index.js]: in my edit event listener I have replaced the prompt with making input field editable when user clicks on edit option user can edit the task. Screen shot attached for your kind reference 
image

File: src/index.js (hunk 1 above)
Comment (baqar-abbas) [reply src/index.js]: For this suggested change I have created another file todo.js and defined my Array there and above function referred by you in the todo.js and imported the module in index.js file. Screen shots attached below for your kind reference. 
image
image

File: src/index.js (hunk 2 above)
Comment (baqar-abbas) [reply src/index.js]: updated please see the comment and screen shot below

File: src/index.js (hunk 3 above)
Comment (baqar-abbas) [reply src/index.js]: updated

**Comment by baqar-abbas** [comment]
All the suggested changes have been incorporated within the project. Sending again for a review

**Review by GedeonTS — APPROVED** [review]
# STATUS: APPROVED :tada: :clap: :champagne: 
Hi @baqar-abbas ,

Your are amazing :smiley: :clap: !  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

 

## Highlights :clap: :tophat: 
- No linter errors :+1: 
- Good use of GitHub flow :+1: 
- Add and Remove are working :+1: 
## Optional suggestions :small_airplane: 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**